### ISSUE TITLE
Schedule readonly appointments should not be deleted with Delete key

### ISSUE THREAD

**Issue by spanmatej** [syncfusion/ej2-angular-ui-components#23]
**Expected behavior:**
According to the documentation, readonly appointments should not be able to be altered by any CRUD action.

**Current behavior:**
On pressing Delete key the normal delete popup shows which enables you to delete the readonly appointment.

I have made the following workaround:
```
public onPopupOpen(args: PopupOpenEventArgs): void {
    // Cancel the delete popup event if the event is readonly
    let data: any = args.data;
    if (args.type == "DeleteAlert") {
      if (data.event["IsReadonly"]) {
        args.cancel = true;
      }
    }
  }
```
`ej2-angular-schedule: 16.4.55`

Can we expect this being added in the future updates, or is this workaround the expected thing to do?

**Comment by karthigeyan92** [comment]
Dear Customer, 
 
We logged the below bug report and the fix will be included in JS2 weekly release which is expected to be rolled out on May 14, 2019. 
https://www.syncfusion.com/feedback/6142/need-to-prevent-the-crud-operations-for-readonly-events

 
Regards, 
Karthi

**Comment by karthigeyan92** [comment]
Dear Customer, 
 
The reported issue has been fixed and included in our weekly patch release (v17.1.48) and same can be checked in the following sample. 
 
https://stackblitz.com/edit/angular-8tzyyq?file=app.component.ts  
 
Regards, 
Karthi

**Comment by Sridhar-Narasimhan** [comment]
Closing the issue due to inactivity for a long time and solution already provided.
